src/agents/customer_group.jl

```diff
@@ -840,57 +860,262 @@ function solve_agent_problem!(
+                regulator.p(z, h, d, t) *
+                min(
+                    customers.d(h, z, d, t) * (1 - utility.loss_dist),
+                    customers.rho_DG(h, :BTMPV, z, d, t) * customers.Opti_DG(z, h, :BTMPV),
+                ) for d in model_data.index_d, t in model_data.index_t
+            ) +
+            # value of distributed generation (excess generation)
+            sum(
+                model_data.omega(d) * delta_t *
+                regulator.p_ex(z, h, d, t) *
+                max(
+                    0,
+                    customers.rho_DG(h, :BTMPV, z, d, t) * customers.Opti_DG(z, h, :BTMPV) -
+                    customers.d(h, z, d, t) * (1 - utility.loss_dist),
+                ) for d in model_data.index_d, t in model_data.index_t
+            ) -
+            # cost of distributed generation 
+            customers.FOM_DG(z, h, :BTMPV) * customers.Opti_DG(z, h, :BTMPV)
+    end
+
+
+
+    for z in model_data.index_z, h in model_data.index_h
+        if BTM_PV_plus_Storage_Only == true
+            if NetProfit(z, h) > 0.0
+                customers.Payback_pv_stor(z, h, :) .=
+                    sum(customers.CapEx_DG(z, h, m) * customers.Opti_DG(z, h, m) for m in customers.index_m) / NetProfit(z, h)
+                # Calculate maximum market share and maximum DG potential (based on WTP curve)
+                customers.MarketShare_pv_stor(z, h, :) .=
+                    1.0 - Distributions.cdf(
+                        Distributions.Gamma(
+                            adopt_model.Shape(z, h, :BTMPV),
+                            1 / adopt_model.Rate(z, h, :BTMPV),
+                        ),
+                        customers.Payback_pv_stor(z, h),
                     )
-                ) / (
-                    1.0 +
-                    (adopt_model.Bass_q(z, h, :BTMPV) / adopt_model.Bass_p(z, h, :BTMPV)) * exp(
-                        -(adopt_model.Bass_p(z, h, :BTMPV) + adopt_model.Bass_q(z, h, :BTMPV)) *
-                        (customers.year_pv(z, h) + 1),
+                customers.MaxDG_pv(z, h, :) .=
+                    customers.MarketShare_pv_stor(z, h) * customers.gamma(z, h) * customers.Opti_DG(z, h, :BTMPV)
+                # Calculate the percentage of existing DER (per agent type per DER technology) as a fraction of maximum DG potential
+                customers.F_pv(z, h, :) .= min(customers.x_DG_E(h, z, :BTMPV) / customers.MaxDG_pv(z, h), 1.0)
+                # Back out the reference year of DER based on the percentage of existing DER
+                customers.year_pv(z, h, :) .=
+                    -log(
+                        (1 - customers.F_pv(z, h)) /
+                        (customers.F_pv(z, h) * adopt_model.Bass_q(z, h, :BTMPV) / adopt_model.Bass_p(z, h, :BTMPV) + 1),
+                    ) / (adopt_model.Bass_p(z, h, :BTMPV) + adopt_model.Bass_q(z, h, :BTMPV))
+                # Calculate incremental DG build
+                customers.A_pv(z, h, :) .=
```

Comment:
Can `A_pv` have a more descriptive name?

Reply:
Not changing for now. Pulled from dGen documentation. Can maybe make a more direct reference.